src/debugger/server/SourceMapManager.js

```diff
@@ -0,0 +1,204 @@
+    // Use presence of directory root to indicate which path format sourcemap prefixes take on.
+    if (directoryRoot !== undefined) {
+      if (sourceMaps === undefined) {
+        throw new DebuggerError(
+          "Invalid input",
+          "Can't provide a sourcemap directory root without having sourcemaps present"
+        );
+      }
+      this._sourcemapDirectoryRoot = directoryRoot;
+      if (this._sourcemapDirectoryRoot[this._sourcemapDirectoryRoot.length - 1] === "/") {
+        // Remove trailing slash to prepare for prepending to internal paths.
+        this._sourcemapDirectoryRoot = this._sourcemapDirectoryRoot.slice(0, -1);
+      }
+    } else {
+      // If sourcemaps don't exist, set prefixes to undefined and break.
+      if (sourceMaps) {
+        for (let map of sourceMaps) {
+          if (map.sourceMapContents === undefined || map.sourceMapContents === "") {
+            this._sourcemapCommonPrefix = undefined;
+            this._sourcemapMapDifference = undefined;
+            return;
+          }
+        }
+      } else {
+        this._sourcemapCommonPrefix = undefined;
+        this._sourcemapMapDifference = undefined;
+        return;
+      }
+
+      // Extract common prefix and map difference
+      let originalSourcePaths = [];
+      let mapPaths = [];
+      for (let map of sourceMaps) {
+        invariant(map.sourceMapContents); // Checked above.
+        let parsed = JSON.parse(map.sourceMapContents);
+        // Two formats for sourcemaps exist.
+        if ("sections" in parsed) {
+          for (let section of parsed.sections) {
+            for (let source of section.map.sources) {
+              originalSourcePaths.push(this._getAbsoluteSourcePath(map.filePath, source));
+            }
+          }
+        } else {
+          for (let source of parsed.sources) {
+            originalSourcePaths.push(this._getAbsoluteSourcePath(map.filePath, source));
+          }
+        }
+        mapPaths.push(this._stripEmptyStringBookends(map.filePath.split("/")));
+      }
+
+      let originalSourceCommonPrefix = this._findCommonPrefix(originalSourcePaths);
+      let originalSourceCPElements = this._stripEmptyStringBookends(originalSourceCommonPrefix.split("/"));
+      let mapCommonPrefix = this._findCommonPrefix(mapPaths);
+      let mapCPElements = this._stripEmptyStringBookends(mapCommonPrefix.split("/"));
+
+      this._sourcemapCommonPrefix = this._findCommonPrefix([originalSourceCPElements, mapCPElements]);
+      this._sourcemapMapDifference = this._findMapDifference(this._sourcemapCommonPrefix, mapCommonPrefix);
+    }
+  }
+
+  // Prefixes used to translate between relative paths stored in AST nodes and absolute paths given to IDE.
+  _sourcemapCommonPrefix: void | string; // For paths relative to map location.
+  _sourcemapMapDifference: void | string; // For paths relative to map location.
+  _sourcemapDirectoryRoot: void | string; // For paths relative to directory root.
```

Comment:
1. In comment, please explicitly call out _sourcemapCommonPrefix/_sourcemapMapDifference are for babel relative path scenario and _sourcemapDirectoryRoot is used by buck scenario.
2. change _sourcemapDirectoryRoot to _buckRoot